### PR TITLE
reliability: add deploy

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**

### CODE REVIEW

File: scripts/deploy_reliability.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+with tempfile.NamedTemporaryFile(mode="w") as fp:
+    for line in [
+        os.getenv("CIRCLE_BRANCH"),
+        os.getenv("CIRCLE_SHA1"),
+        os.getenv("DDTRACE_VERSION"),
+        os.getenv("CIRCLE_USERNAME"),
+    ]:
+        fp.write(f"{line}\n")
+    fp.seek(0)
```
Comment (jd) [inline scripts/deploy_reliability.py:21]:
```suggestion
```
I don't see why this is necessary?

**Comment by brettlangdon** [comment]
We instead from wheels, so we have to trigger something post building/publishing our wheels.

File: .circleci/config.yml [hunk 2]
```diff
@@ -820,6 +831,12 @@ workflows:
       - build_docs
       - black
       - flake8
+      - deploy_reliability:
```
Comment (brettlangdon) [inline .circleci/config.yml:834]: this should require `deploy_master`, we want to do it after we successfully publish the wheels.

File: scripts/deploy_reliability.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply scripts/deploy_reliability.py:21]: Copy-paste from the given script 🤷‍♂

File: .circleci/config.yml (hunk 2 above)
Comment (Kyle-Verhoog) [reply .circleci/config.yml:834]: I don't see why this is necessary, we can (and do) install using the commit hash

```
pip install git+https://github.com/DataDog/dd-trace-py.git@<HASH>
```

**Comment by Kyle-Verhoog** [comment]
Agreed, but this is the system that's in place 🤷‍♂

File: .circleci/config.yml (hunk 2 above)
Comment (brettlangdon) [reply .circleci/config.yml:834]: if we do it from a wheel, we can verify that our wheels building process works, and it won't require Cython to be installed in order to build/deploy into the reliability env.

Comment (Kyle-Verhoog) [reply .circleci/config.yml:834]: Got it, but if we want to test that our wheels work the reliability env is (currently) not the place to do it. We don't test every python version or OS platform. Perhaps we want to investigate adding a job in our CI to test the wheels? cc @majorgreys

Comment (Kyle-Verhoog) [reply .circleci/config.yml:834]: Also how could we permit deploying non-master, non-staging branches which don't have wheels built?

Comment (brettlangdon) [reply .circleci/config.yml:834]: got it. your call here then

File: .circleci/config.yml [hunk 3]
```diff
@@ -820,6 +831,12 @@ workflows:
...
+          filters:
+            branches:
+              only:
+                - master
+                - /.*-reliability/
```
Comment (brettlangdon) [inline .circleci/config.yml:839]: should we just do like `master` and `staging`?

Comment (Kyle-Verhoog) [reply .circleci/config.yml:839]: Thought here was that I was hoping to be able to push my performance testing changes to a reliability branch to be able to make verifying easier. 🙂

**Pull request closed without merging**